src/integration_tests/mission_transition.cpp

```diff
@@ -0,0 +1,151 @@
+#include <iostream>
+#include <functional>
+#include <memory>
+#include <future>
+#include <atomic>
+#include <cmath>
+#include "integration_test_helper.h"
+#include "path_checker.h"
+#include "mavsdk.h"
+#include "plugins/telemetry/telemetry.h"
+#include "plugins/action/action.h"
+#include "plugins/mission/mission.h"
+
+using namespace mavsdk;
+using namespace std::placeholders; // for `_1`
+
+void do_mission_takeoff_transitions_land(float mission_altitude_m);
+
+TEST_F(SitlTest, MissionTakeoffAndLandHigh)
+{
+    do_mission_takeoff_transitions_land(20);
+}
+
+TEST_F(SitlTest, MissionTakeoffAndLandLow)
+{
+    do_mission_takeoff_transitions_land(5);
+}
+
+void do_mission_takeoff_transitions_land(float mission_altitude_m)
+{
+    Mavsdk mavsdk;
+
+    {
+        auto prom = std::make_shared<std::promise<void>>();
+        auto future_result = prom->get_future();
+
+        LogInfo() << "Waiting to discover system...";
+        mavsdk.subscribe_on_new_system([&mavsdk, prom]() {
+            const auto system = mavsdk.systems().at(0);
+
+            if (system->is_connected()) {
+                LogInfo() << "Discovered system";
+                prom->set_value();
+            }
+        });
+
+        ConnectionResult ret = mavsdk.add_udp_connection();
+        ASSERT_EQ(ret, ConnectionResult::Success);
+
+        auto status = future_result.wait_for(std::chrono::seconds(2));
+        ASSERT_EQ(status, std::future_status::ready);
+        future_result.get();
+        // FIXME: This hack is to prevent that the promise is set twice.
+        mavsdk.subscribe_on_new_system(nullptr);
+    }
+
+    auto system = mavsdk.systems().at(0);
+    auto telemetry = std::make_shared<Telemetry>(system);
+    auto mission = std::make_shared<Mission>(system);
+    auto action = std::make_shared<Action>(system);
+
+    PathChecker pc;
+
+    telemetry->subscribe_position([&pc](Telemetry::Position position) {
+        pc.check_current_alitude(position.relative_altitude_m);
+    });
+
+    while (!telemetry->health_all_ok()) {
+        LogInfo() << "Waiting for system to be ready";
+        LogDebug() << "Health: " << telemetry->health();
+        std::this_thread::sleep_for(std::chrono::seconds(1));
+    }
+
+    LogInfo() << "System ready";
+    LogInfo() << "Creating and uploading mission";
+
+    Mission::MissionItem new_item{};
+
+    new_item.latitude_deg = 47.398170327054473;
+    new_item.longitude_deg = 8.5456490218639658;
+    new_item.relative_altitude_m = mission_altitude_m;
+    new_item.vehicle_action = Mission::MissionItem::VehicleAction::Takeoff;
+
+    Mission::MissionPlan mission_plan{};
+    mission_plan.mission_items.push_back(new_item);
+
+    new_item.vehicle_action = Mission::MissionItem::VehicleAction::TransitionToFw;
```

Comment:
What's the expected behavior when flying a multicopter? I guess it is ignored, isn't it?

Reply:
I guess so. I was thinking, that if landing and takeoff would work, this action will provide that information.